tensorflow/c/kernels.cc

```diff
@@ -44,8 +45,11 @@ TF_KernelBuilder* TF_NewKernelBuilder(
     void (*compute_func)(void*, TF_OpKernelContext*),
     void (*delete_func)(void*)) {
   TF_KernelBuilder* result = new TF_KernelBuilder;
+  const char* subdevice_name =
+      ::tensorflow::DeviceFactory::SubDeviceType(device_name).c_str();
```

Comment:
I think this is doing a use-after-free -- the `string` returned by `SubDeviceType` will be immediately freed I believe.

Reply:
Oh, yes, you reminder me, originally I store it to the stack and then pass it to the following function. does it acceptable or you have another suggestion?
```
  -const std::string subdevice_type =
  +const char* subdevice_name =
      ::tensorflow::DeviceFactory::SubDeviceType(device_name).c_str();
  -const char* subdevice_name = subdevice_type.c_str();
```

Reply:
done,  sorry i miss understanding the comment.